include/odp/api/spec/queue_types.h

```diff
@@ -291,6 +301,29 @@ typedef struct odp_queue_param_t {
 	  * default size. The default value is 0. */
 	uint32_t size;
 
+	/** Number of vector aggregators
+	  *
+	  * When >= 1, event vector generation is enabled and configured
+	  * according to vector configuration parameters. Otherwise, vector
+	  * configuration parameters are ignored. The default value is 0.
+	  *
+	  * Depending on queue type, vector aggregation capabilities can be read
+	  * either from odp_queue_capability_t::plain.vector or
+	  * odp_schedule_capability_t::vector. */
+	uint32_t num_vector;
+
+	/** Vector configuration parameters
+	  *
+	  * When event vector generation is enabled, events enqueued to this
+	  * queue are by default vectorized using vector aggregator 0. When
+	  * 'num_vector' >1, other vector aggregators can be utilized with
+	  * odp_queue_enq_vector() function.
+	  *
+	  * When vector generation is enabled, events may be delivered both as
+	  * event vector events and normal events. Implementation won't split up
+	  * event vectors successfully enqueued by the application. */
+	odp_event_vector_config_t vector[ODP_QUEUE_MAX_VECTOR_AGGR];
```

Comment:
Other option could be to create vector generator profile, and give handle to that here instead of full set of config params ? That would enable easy reuse of the same set of params for many queues (that have the same aggregation needs).